contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -43,42 +43,39 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
+    uint256 public messageNonce = 1;
 
     // Tracks the outstanding receipts to send back to a given chain in subsequent messages sent to that chain.
-    // Key is the blockchain ID of the other chain, and the value is a queue of pending receipts for messages
+    // The key is the blockchain ID of the other chain, and the value is a queue of pending receipts for messages
     // received from that chain.
     mapping(bytes32 sourceBlockchainID => ReceiptQueue.TeleporterMessageReceiptQueue receiptQueue)
         public receiptQueues;
 
     // Tracks the message hash and fee information for each message sent that has yet to be acknowledged
-    // with a receipt. The messages are tracked per chain and keyed by message ID.
-    // The first key is the blockchain ID of the destination chain, the second key is the message ID, and the value is the info
-    // for the uniquely identified message.
-    mapping(
-        bytes32 destinationBlockchainID => mapping(uint256 messageID => SentMessageInfo messageInfo)
-    ) public sentMessageInfo;
+    // with a receipt. The key is the message ID, and the value is the info for the uniquely identified message.
+    mapping(bytes32 messageID => SentMessageInfo messageInfo) public sentMessageInfo;
 
-    // Tracks the hash of messages that have been received but whose execution has never succeeded.
+    // Tracks the hash of messages that have been received but have never succeeded in execution.
     // Enables retrying of failed messages with higher gas limits. Message execution is guaranteed to
-    // succeed at most once. The first key is the blockchain ID of the sending chain, the second key is the message ID, and
-    // the value is the hash of the uniquely identified message whose execution failed.
-    mapping(bytes32 sourceBlockchainID => mapping(uint256 messageID => bytes32 messageHash)) public
-        receivedFailedMessageHashes;
-
-    // Tracks the relayer reward address for each message delivered from a given chain.
-    // Note that these values are also used to determine if a given message has been delivered or not.
-    // The first key is the blockchain ID, the second key is the message ID, and the value is the reward address
-    // provided by the deliverer of the uniquely identified message.
-    mapping(
-        bytes32 sourceBlockchainID => mapping(uint256 messageID => address relayerRewardAddress)
-    ) internal _relayerRewardAddresses;
+    // succeed at most once. The key is the message ID, and the value is the hash of the uniquely
+    // identified message whose execution failed.
+    mapping(bytes32 messageID => bytes32 messageHash) public receivedFailedMessageHashes;
+
+    // Tracks the message nonce for each message that has been received. The key is the message ID,
+    // and the value is the nonce value that was received as a part for that message.
+    // Note: the `messageNonce` values are also used to determine if a given message has been received or not.
```

Comment:
Should a slightly more extended explanation of the messageNonce be provided here? This might be the first point that someone comes across the term messageNonce.

Reply:
I think the comment above for `messageNonce` on lines 50-53 explain it in more detail, but happy to expand on it there if you think there's a piece that's missing.